chore: remove anonymous feature

patric-dosch-vi opened the pull request.

patric-dosch-vi committed:
chore: remove anonymous feature

Refs: CARITAS-243

vi-sudo commented:
Task linked: [CARITAS-243 [Caritas Rework] WP2.1: Remove legacy code](https://app.clickup.com/t/6656196/CARITAS-243)

patric-dosch-vi committed:
chore: remove anonymous feature

Two tests aren't running in the workflow; we'll disable them for now, as they only test "loggers."

- SetRocketChatRoomReadOnlyActionCommandTest.execute_Should_logError_When_rocketChatCallFails:69
- InactivePrivateGroupsProviderTest.retrieveUserWithInactiveGroupsMap_ShouldLogError_WhenFetchOfInactiveGroupsFails:101

Refs: CARITAS-243
